src/optimization/design_parameterization/inner_vol_parameterization.cpp

```diff
@@ -0,0 +1,157 @@
+#include "inner_vol_parameterization.hpp"
+#include <deal.II/dofs/dof_tools.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/sparsity_tools.h>
+
+namespace PHiLiP {
+
+template<int dim>
+DesignParameterizationInnerVol<dim> :: DesignParameterizationInnerVol(
+    std::shared_ptr<HighOrderGrid<dim,double>> _high_order_grid)
+    : DesignParameterizationBase<dim>(_high_order_grid)
+{
+    compute_inner_vol_index_to_vol_index();
+}
+
+template<int dim>
+void DesignParameterizationInnerVol<dim> :: compute_inner_vol_index_to_vol_index()
+{
+    unsigned int n_vol_nodes = this->high_order_grid->volume_nodes.size();
+    unsigned int n_surf_nodes = this->high_order_grid->surface_nodes.size();
+    n_inner_nodes = n_vol_nodes - n_surf_nodes;
+
+    dealii::LinearAlgebra::distributed::Vector<int> is_a_surface_node;
```

Comment:
bool gives compilation error (probably because dealii's vector template needs a [Number](https://www.dealii.org/current/doxygen/deal.II/classVector.html) type.

Reply:
Sounds good! `int` behaves perfectly well in this scenario, and I didn't know that the vector template needs a number.